source/common/network/socket_option_factory.cc

```diff
@@ -33,27 +33,55 @@ SocketOptionFactory::buildTcpKeepaliveOptions(Network::TcpKeepaliveConfig keepal
 std::unique_ptr<Socket::Options> SocketOptionFactory::buildIpFreebindOptions() {
   std::unique_ptr<Socket::Options> options = absl::make_unique<Socket::Options>();
   options->push_back(std::make_shared<Network::AddrFamilyAwareSocketOptionImpl>(
-      Network::Socket::SocketState::PreBind, ENVOY_SOCKET_IP_FREEBIND, ENVOY_SOCKET_IPV6_FREEBIND,
-      1));
+      envoy::api::v2::core::SocketOption::STATE_PREBIND, ENVOY_SOCKET_IP_FREEBIND,
+      ENVOY_SOCKET_IPV6_FREEBIND, 1));
   return options;
 }
 
 std::unique_ptr<Socket::Options> SocketOptionFactory::buildIpTransparentOptions() {
   std::unique_ptr<Socket::Options> options = absl::make_unique<Socket::Options>();
   options->push_back(std::make_shared<Network::AddrFamilyAwareSocketOptionImpl>(
-      Network::Socket::SocketState::PreBind, ENVOY_SOCKET_IP_TRANSPARENT,
+      envoy::api::v2::core::SocketOption::STATE_PREBIND, ENVOY_SOCKET_IP_TRANSPARENT,
       ENVOY_SOCKET_IPV6_TRANSPARENT, 1));
   options->push_back(std::make_shared<Network::AddrFamilyAwareSocketOptionImpl>(
-      Network::Socket::SocketState::PostBind, ENVOY_SOCKET_IP_TRANSPARENT,
+      envoy::api::v2::core::SocketOption::STATE_BOUND, ENVOY_SOCKET_IP_TRANSPARENT,
       ENVOY_SOCKET_IPV6_TRANSPARENT, 1));
   return options;
 }
 
+std::unique_ptr<Socket::Options> SocketOptionFactory::buildLiteralOptions(
```

Comment:
I only see this being called for listener socket options.

Reply:
Oops. The description was supposed to say "listener only, upstream to follow." I've got the upstream change staged, just wanted to keep things smaller for fast turnaround.